src/transaction/types/utxo_id.rs

```diff
@@ -0,0 +1,113 @@
+use std::str::FromStr;
+
+use fuel_types::Bytes32;
+
+use crate::TxId;
+
+/// Identification of unspend transaction output.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(
+    feature = "serde-types-minimal",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct UtxoId {
+    /// transaction id
+    pub tx_id: TxId,
+    /// output index
+    pub output_index: u8,
+}
+
+impl core::fmt::LowerHex for UtxoId {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        if f.alternate() {
+            write!(f, "{:#x}{:02x}", self.tx_id, self.output_index)
+        } else {
+            write!(f, "{:x}{:02x}", self.tx_id, self.output_index)
+        }
+    }
+}
+
+impl core::fmt::UpperHex for UtxoId {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        if f.alternate() {
+            write!(f, "{:#X}{:02X}", self.tx_id, self.output_index)
+        } else {
+            write!(f, "{:X}{:02X}", self.tx_id, self.output_index)
+        }
+    }
+}
+
+impl FromStr for UtxoId {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        const ERR: &str = "Invalid encoded byte";
+        let alternate = s.starts_with("0x");
+        let s = if alternate { &s[2..] } else { s };
+        let utxo_id = if s.len() == 0 {
+            UtxoId {
+                tx_id: Bytes32::default(),
+                output_index: 0,
+            }
+        } else {
+            UtxoId {
+                tx_id: Bytes32::from_str(&s[..s.len() - 1])?,
+                output_index: s
+                    .as_bytes()
+                    .last()
+                    .cloned()
+                    .map(hex_val)
+                    .flatten()
+                    .ok_or(ERR)?,
+            }
+        };
+        Ok(utxo_id)
+    }
+}
+
+const fn hex_val(c: u8) -> Option<u8> {
```

Comment:
Duplicated code to `fuel-types`. Should not be here after `TxId` is moved there

Reply:
Removed duplicated part. Found that there is `u8::from_str_radix` that does the job.